parsec/core/fs/userfs.py

```diff
@@ -0,0 +1,140 @@
+# Parsec Cloud (https://parsec.cloud) Copyright (c) AGPLv3 2019 Scille SAS
+
+import inspect
+from typing import Dict
+from uuid import UUID
+
+from parsec.types import UserID
+from parsec.event_bus import EventBus
+from parsec.core.types import LocalDevice, FsPath
+from parsec.core.local_storage import LocalStorage
+from parsec.core.backend_connection import BackendCmdsPool
+from parsec.core.fs.local_folder_fs import (
+    FSManifestLocalMiss,
+    FSMultiManifestLocalMiss,
+    LocalFolderFS,
+)
+from parsec.core.fs.local_file_fs import LocalFileFS, FSBlocksLocalMiss
+from parsec.core.fs.syncer import Syncer
+from parsec.core.fs.sharing import Sharing
+from parsec.core.fs.remote_loader import RemoteLoader
+from parsec.core.fs.workspacefs import WorkspaceFS
+from parsec.core.remote_devices_manager import RemoteDevicesManager
+
+
+class UserFS:
+    def __init__(
+        self,
+        device: LocalDevice,
+        local_storage: LocalStorage,
+        backend_cmds: BackendCmdsPool,
+        remote_devices_manager: RemoteDevicesManager,
+        event_bus: EventBus,
+    ):
+        self.device = device
+        self.local_storage = local_storage
+        self.backend_cmds = backend_cmds
+        self.event_bus = event_bus
+
+        self._local_folder_fs = LocalFolderFS(device, local_storage, event_bus)
+        self._local_file_fs = LocalFileFS(device, local_storage, self._local_folder_fs, event_bus)
+        self._remote_loader = RemoteLoader(backend_cmds, remote_devices_manager, local_storage)
+        self._syncer = Syncer(
+            device,
+            backend_cmds,
+            remote_devices_manager,
+            self._local_folder_fs,
+            self._local_file_fs,
+            event_bus,
+        )
```

Comment:
I think we'll need to have all the following objects instanciated once for each workspace fs:
- local folder fs
- local file fs
- syncer

Not sure about the remote loader though.